drivers/node/ssh/ssh.go

```diff
@@ -466,31 +466,39 @@ func (s *SSH) doCmdUsingPodWithoutRetry(n node.Node, cmd string) (string, error)
 }
 
 func (s *SSH) doCmdUsingPod(n node.Node, options node.ConnectionOpts, cmd string, ignoreErr bool) (string, error) {
-	cmds := []string{"nsenter", "--mount=/hostproc/1/ns/mnt", "/bin/bash", "-c", cmd}
-
-	allPodsForNode, err := k8sCore.GetPodsByNode(n.Name, execPodDefaultNamespace)
-	if err != nil {
-		logrus.Errorf("failed to get pods in node: %s err: %v", n.Name, err)
-		return "", err
-	}
 	var debugPod *v1.Pod
-	for _, pod := range allPodsForNode.Items {
-		if pod.Labels["name"] == execPodDaemonSetLabel && k8sCore.IsPodReady(pod) {
-			debugPod = &pod
-			break
+
+	findDebugPodFunc := func() (interface{}, bool, error) {
+		allPodsForNode, err := k8sCore.GetPodsByNode(n.Name, execPodDefaultNamespace)
+		if err != nil {
+			logrus.Errorf("failed to get pods in node: %s err: %v", n.Name, err)
+			return nil, true, err
+		}
+		for _, pod := range allPodsForNode.Items {
+			if pod.Labels["name"] == execPodDaemonSetLabel && k8sCore.IsPodReady(pod) {
+				debugPod = &pod
+				break
+			}
 		}
-	}
 
-	if debugPod == nil {
-		return "", &node.ErrFailedToRunCommand{
-			Node:  n,
-			Cause: fmt.Sprintf("debug pod not found in node %v", n),
+		if debugPod == nil {
+			return nil, true, &node.ErrFailedToRunCommand{
+				Node:  n,
+				Cause: fmt.Sprintf("debug pod not found in node %v", n),
+			}
 		}
+		return nil, false, nil
+	}
+
+	logrus.Debugf("Finding the debug pod to run command on node %s", n.Name)
```

Comment:
why not run the findDebugPod and RunCommand in the same task?

Reply:
Once the debug pod is ready after reboot, there is no need to find it again every time we retry the specified command. But I guess we can avoid that by checking if debugPod is not nil. Will change.